enclave_build/docs/cache.md

```diff
@@ -0,0 +1,127 @@
+      "size": 425,
+      "digest": "sha256:98806831de537d56f19b4b3cbf6ed80187374a1521a18d2d0c5f8a0e3962b2ae",
+      "annotations": {
+        "org.opencontainers.image.ref.name": "docker.io/test-oci:hello"
+      }
+    }
+  ]
+}
+```
+
+As we can see, the index holds one manifest entry, for one image stored. The number of manifests increases with each newly stored image. The manifest hash will point us to the manifest blob.
+
+#### Layout file
+
+oci-layout:
+```
+{"imageLayoutVersion":"1.0.0"}
+```
+
+Will simply define the OCI image layout version we are using.
+
+#### Manifest
+
+98806831de537d56f19b4b3cbf6ed80187374a1521a18d2d0c5f8a0e3962b2ae
+```
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+  "config": {
+    "mediaType": "application/vnd.docker.container.image.v1+json",
+    "digest": "sha256:feb5d9fea6a5e9606aa995e879d862b825965ba48de054caab5ef356dc6b3412",
+    "size": 1469
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.docker.image.rootfs.diff.tar.gzip",
+      "digest": "sha256:2db29710123e3e53a794f2694094b9b4338aa9ee5c40b930cb8063a1be392c54",
+      "size": 2479
+    }
+  ]
+}
+```
+
+The manifest is our guide for a particular image. It tells us which components among the blobs belong to the image it defines. It will tell us the digest of the content for the config file and each layer. With this information we can search the blobs.
+
+#### Config
+
+The configuration file will define the behavior of the image inside the container, platform dependencies and interaction with the filesystem composed from the layers. Nitro CLI's usage of the cache should not concern with the contents of this file (excepting UTF-8 and digest validation).
+
+#### Layers
+
+Layers are serialized filesystem changes that go on top of each other, starting from the base filesystem we create the image with. The file will be binary content that, from Nitro CLI's point of view, just has to obey the content digest in the manifest.
+
+## Operations
+
+The operations offered by the `cache` module are as follows:
+
+### Create
+
+When building our first image, the cache will probably not exist. This operation will create, prior to storing an image, an empty cache. It will contain the cache root folder, an index file, a blobs folder and the oci-layout.
+
+### Store
+
+This is the operation which adds an image to the cache. It will be called on an empty cache or after a cache miss when searching for the image. After pulling the image and extracting its details, its manifest entry will be added to the index and the blobs will be written, each to their file.
+
+### Validate
+
+After we find an image in the index or the in-memory cache, we want to validate its structure before using the data. The validation starts from the index file. It should contain our image's reference and the digest of the manifest. If fields are missing or the JSON structure does not comply, an error will be returned.
```

Comment:
Is the in-memory cache the same you are talking about until this point or are we discussing about two different caches?

Reply:
The cache itself is made up of the file structure presented. The in-memory cache is what the `CacheManager` loads (the map: image reference -> config digest). I will clarify.